The job **`x86_64-gnu-llvm-14`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5008164355/jobs/8975713134) [(plain)](https://github.com/rust-lang/rust/commit/0ff3d0bfcd5e138cae2dee96ae113ba8ca2b628a/checks/13565029687/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
running 465 tests
i.i...i.i..ii............i....i..ii.................iii........ii.i.i............i......  88/465
..............ii..................i..............i.....i........i..iiii....iiiiiiii....i 176/465
..iiiii........ii.i....i.i.iiii.iiii.....ii....i.ii......i..i.......................i... 264/465
ii..i.i.....ii..i.ii..............ii.i...........................ii......F.....F........ 352/465
iiiiii...................

failures:


---- [codegen] tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs#aarch64 stdout ----

error in revision `aarch64`: verification with 'FileCheck' failed
status: exit status: 1
command: "/usr/lib/llvm-14/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.aarch64/sanitizer-kcfi-emit-type-metadata-trait-objects.ll" "/checkout/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs" "--allow-unused-prefixes" "--check-prefixes" "CHECK,NONMSVC,aarch64" "--dump-input-context" "100"
stdout: none
/checkout/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs:99:12: error: CHECK: expected string not found in input
/checkout/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs:99:12: error: CHECK: expected string not found in input
 // CHECK: call void %{{[0-9]}}({{\{\}\*|ptr}} align 1 {{%[a-z]\.0|%_[0-9]}}, {{\{\}\*|ptr}} align 1 {{%[a-z]\.0|%_[0-9]}}){{.*}}[ "kcfi"(i32 [[TYPE3:[[:print:]]+]]) ]
           ^
           ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.aarch64/sanitizer-kcfi-emit-type-metadata-trait-objects.ll:105:183: note: scanning from here
define dso_local void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo317hdb324cf16b626e72E({}* align 1 %a.0, [3 x i64]* align 8 %a.1) unnamed_addr #1 !<unknown kind #36> !6 {
/checkout/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs:108:12: error: CHECK: expected string not found in input
/checkout/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs:108:12: error: CHECK: expected string not found in input
 // CHECK: call void %{{[0-9]}}({{\{\}\*|ptr}} align 1 {{%[a-z]\.0|%_[0-9]}}, {{\{\}\*|ptr}} align 1 {{%[a-z]\.0|%_[0-9]}}){{.*}}[ "kcfi"(i32 [[TYPE3:[[:print:]]+]]) ]
           ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.aarch64/sanitizer-kcfi-emit-type-metadata-trait-objects.ll:117:142: note: scanning from here
define dso_local void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4bar317hd298b3b9b9d428c0E() unnamed_addr #1 !<unknown kind #36> !1 {


Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.aarch64/sanitizer-kcfi-emit-type-metadata-trait-objects.ll


-dump-input=help explains the following input dump.
Input was:
<<<<<<
           .
           .
           .
           .
           5:  
           6: %Type1 = type {} 
           7: %Type2 = type {} 
           8: %Type3 = type {} 
           9:  
          10: @vtable.0 = private unnamed_addr constant <{ i8*, [16 x i8], i8* }> <{ i8* bitcast (void ()* @"_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects78drop_in_place_fn$LT$sanitizer_kcfi_emit_type_metadata_trait_objects..Type1$GT$17hd544d746556e563bE" to i8*), [16 x i8] c"\00\00\00\00\00\00\00\00\01\00\00\00\00\00\00\00", i8* bitcast (void (%Type1*)* @"_ZN130_$LT$sanitizer_kcfi_emit_type_metadata_trait_objects..Type1$u20$as$u20$sanitizer_kcfi_emit_type_metadata_trait_objects..Trait1$GT$3foo17h3ed069aedb940e54E" to i8*) }>, align 8 
          11: @vtable.1 = private unnamed_addr constant <{ i8*, [16 x i8], i8* }> <{ i8* bitcast (void ()* @"_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects78drop_in_place_fn$LT$sanitizer_kcfi_emit_type_metadata_trait_objects..Type2$GT$17h8aa09b16cfc8ee65E" to i8*), [16 x i8] c"\00\00\00\00\00\00\00\00\01\00\00\00\00\00\00\00", i8* bitcast (void (%Type2*)* @"_ZN141_$LT$sanitizer_kcfi_emit_type_metadata_trait_objects..Type2$u20$as$u20$sanitizer_kcfi_emit_type_metadata_trait_objects..Trait2$LT$i32$GT$$GT$3bar17hc43b8b747eac5e80E" to i8*) }>, align 8 
          12: @vtable.2 = private unnamed_addr constant <{ i8*, [16 x i8], i8* }> <{ i8* bitcast (void ()* @"_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects78drop_in_place_fn$LT$sanitizer_kcfi_emit_type_metadata_trait_objects..Type3$GT$17h3b9500d3f54315f5E" to i8*), [16 x i8] c"\00\00\00\00\00\00\00\00\01\00\00\00\00\00\00\00", i8* bitcast (void (%Type3*, %Type3*)* @"_ZN86_$LT$T$u20$as$u20$sanitizer_kcfi_emit_type_metadata_trait_objects..Trait3$LT$U$GT$$GT$3baz17h3db20bfe4fcf6962E" to i8*) }>, align 8 
          13:  
          14: ; sanitizer_kcfi_emit_type_metadata_trait_objects::drop_in_place_fn<sanitizer_kcfi_emit_type_metadata_trait_objects::Type1> 
          15: ; Function Attrs: inlinehint noredzone nounwind 
          16: define internal void @"_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects78drop_in_place_fn$LT$sanitizer_kcfi_emit_type_metadata_trait_objects..Type1$GT$17hd544d746556e563bE"() unnamed_addr #0 !<unknown kind #36> !1 { 
          18:  ret void 
          19: } 
          20:  
          20:  
          21: ; sanitizer_kcfi_emit_type_metadata_trait_objects::drop_in_place_fn<sanitizer_kcfi_emit_type_metadata_trait_objects::Type2> 
          22: ; Function Attrs: inlinehint noredzone nounwind 
          23: define internal void @"_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects78drop_in_place_fn$LT$sanitizer_kcfi_emit_type_metadata_trait_objects..Type2$GT$17h8aa09b16cfc8ee65E"() unnamed_addr #0 !<unknown kind #36> !1 { 
          25:  ret void 
          26: } 
          27:  
          27:  
          28: ; sanitizer_kcfi_emit_type_metadata_trait_objects::drop_in_place_fn<sanitizer_kcfi_emit_type_metadata_trait_objects::Type3> 
          29: ; Function Attrs: inlinehint noredzone nounwind 
          30: define internal void @"_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects78drop_in_place_fn$LT$sanitizer_kcfi_emit_type_metadata_trait_objects..Type3$GT$17h3b9500d3f54315f5E"() unnamed_addr #0 !<unknown kind #36> !1 { 
          32:  ret void 
          33: } 
          34:  
          34:  
          35: ; <sanitizer_kcfi_emit_type_metadata_trait_objects::Type1 as sanitizer_kcfi_emit_type_metadata_trait_objects::Trait1>::foo 
          36: ; Function Attrs: noredzone nounwind 
          37: define dso_local void @"_ZN130_$LT$sanitizer_kcfi_emit_type_metadata_trait_objects..Type1$u20$as$u20$sanitizer_kcfi_emit_type_metadata_trait_objects..Trait1$GT$3foo17h3ed069aedb940e54E"(%Type1* align 1 %self) unnamed_addr #1 !<unknown kind #36> !2 { 
          39:  ret void 
          40: } 
          41:  
          41:  
          42: ; <sanitizer_kcfi_emit_type_metadata_trait_objects::Type2 as sanitizer_kcfi_emit_type_metadata_trait_objects::Trait2<i32>>::bar 
          43: ; Function Attrs: noredzone nounwind 
          44: define dso_local void @"_ZN141_$LT$sanitizer_kcfi_emit_type_metadata_trait_objects..Type2$u20$as$u20$sanitizer_kcfi_emit_type_metadata_trait_objects..Trait2$LT$i32$GT$$GT$3bar17hc43b8b747eac5e80E"(%Type2* align 1 %self) unnamed_addr #1 !<unknown kind #36> !3 { 
          46:  ret void 
          47: } 
          48:  
          48:  
          49: ; <T as sanitizer_kcfi_emit_type_metadata_trait_objects::Trait3<U>>::baz 
          50: ; Function Attrs: noredzone nounwind 
          51: define dso_local void @"_ZN86_$LT$T$u20$as$u20$sanitizer_kcfi_emit_type_metadata_trait_objects..Trait3$LT$U$GT$$GT$3baz17h3db20bfe4fcf6962E"(%Type3* align 1 %self, %Type3* align 1 %_2) unnamed_addr #1 !<unknown kind #36> !4 { 
          53:  ret void 
          54: } 
          55:  
          55:  
          56: ; sanitizer_kcfi_emit_type_metadata_trait_objects::foo1 
          57: ; Function Attrs: noredzone nounwind 
          58: define dso_local void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo117h93502349188b3e24E({}* align 1 %a.0, [3 x i64]* align 8 %a.1) unnamed_addr #1 !<unknown kind #36> !2 { 
          59: start: 
          60:  %0 = bitcast [3 x i64]* %a.1 to void ({}*)** 
          61:  %1 = getelementptr inbounds void ({}*)*, void ({}*)** %0, i64 3 
          62:  %2 = load void ({}*)*, void ({}*)** %1, align 8, !invariant.load !5, !nonnull !5 
          63:  call void %2({}* align 1 %a.0) #2 [ "kcfi"(i32 -56908610) ] 
          64:  ret void 
          65: } 
          66:  
          67: ; sanitizer_kcfi_emit_type_metadata_trait_objects::bar1 
          68: ; Function Attrs: noredzone nounwind 
          69: define dso_local void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4bar117had3ab10c0b90936dE() unnamed_addr #1 !<unknown kind #36> !1 { 
          70: start: 
          71:  %_1 = alloca %Type1, align 1 
          72:  %b.0 = bitcast %Type1* %_1 to {}* 
          73:  %0 = load void ({}*)*, void ({}*)** getelementptr inbounds (void ({}*)*, void ({}*)** bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.0 to void ({}*)**), i64 3), align 8, !invariant.load !5, !nonnull !5 
          74:  call void %0({}* align 1 %b.0) #2 [ "kcfi"(i32 -56908610) ] 
          75:  ret void 
          76: } 
          77:  
          78: ; sanitizer_kcfi_emit_type_metadata_trait_objects::foo2 
          79: ; Function Attrs: noredzone nounwind 
          80: define dso_local void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo217h8161e42dd83c0851E({}* align 1 %a.0, [3 x i64]* align 8 %a.1) unnamed_addr #1 !<unknown kind #36> !3 { 
          81: start: 
          82:  %0 = bitcast [3 x i64]* %a.1 to void ({}*)** 
          83:  %1 = getelementptr inbounds void ({}*)*, void ({}*)** %0, i64 3 
          84:  %2 = load void ({}*)*, void ({}*)** %1, align 8, !invariant.load !5, !nonnull !5 
          85:  call void %2({}* align 1 %a.0) #2 [ "kcfi"(i32 102973893) ] 
          86:  ret void 
          87: } 
          88:  
          89: ; sanitizer_kcfi_emit_type_metadata_trait_objects::bar2 
          90: ; Function Attrs: noredzone nounwind 
          91: define dso_local void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4bar217h18ebf9f7b2e09ed4E() unnamed_addr #1 !<unknown kind #36> !1 { 
          92: start: 
          93:  %_1 = alloca %Type2, align 1 
          94:  %_3.0 = bitcast %Type2* %_1 to {}* 
          95: ; call sanitizer_kcfi_emit_type_metadata_trait_objects::foo2 
          96:  call void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo217h8161e42dd83c0851E({}* align 1 %_3.0, [3 x i64]* align 8 bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.1 to [3 x i64]*)) #2 
          97:  %b.0 = bitcast %Type2* %_1 to {}* 
          98:  %0 = load void ({}*)*, void ({}*)** getelementptr inbounds (void ({}*)*, void ({}*)** bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.1 to void ({}*)**), i64 3), align 8, !invariant.load !5, !nonnull !5 
          99:  call void %0({}* align 1 %b.0) #2 [ "kcfi"(i32 102973893) ] 
         100:  ret void 
         101: } 
         102:  
         103: ; sanitizer_kcfi_emit_type_metadata_trait_objects::foo3 
         104: ; Function Attrs: noredzone nounwind 
         105: define dso_local void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo317hdb324cf16b626e72E({}* align 1 %a.0, [3 x i64]* align 8 %a.1) unnamed_addr #1 !<unknown kind #36> !6 { 
check:99                                                                                                                                                                                            X~~ error: no match found
check:99      ~~~~~~~
check:99      ~~~~~~~
         107:  %_2 = alloca %Type3, align 1 
check:99      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         108:  %0 = bitcast [3 x i64]* %a.1 to void ({}*, %Type3*)** 
check:99      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         109:  %1 = getelementptr inbounds void ({}*, %Type3*)*, void ({}*, %Type3*)** %0, i64 3 
check:99      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         110:  %2 = load void ({}*, %Type3*)*, void ({}*, %Type3*)** %1, align 8, !invariant.load !5, !nonnull !5 
check:99      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         111:  call void %2({}* align 1 %a.0, %Type3* align 1 %_2) #2 [ "kcfi"(i32 -1903783964) ] 
         112:  ret void 
check:99      ~~~~~~~~~~
         113: } 
check:99      ~~
check:99      ~~
         114:  
check:99      ~
         115: ; sanitizer_kcfi_emit_type_metadata_trait_objects::bar3 
check:99      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         116: ; Function Attrs: noredzone nounwind 
check:99      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         117: define dso_local void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4bar317hd298b3b9b9d428c0E() unnamed_addr #1 !<unknown kind #36> !1 { 
check:99      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:108                                                                                                                                                  X~~ error: no match found
check:108     ~~~~~~~
check:108     ~~~~~~~
         119:  %_1 = alloca %Type3, align 1 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         120:  %_3.0 = bitcast %Type3* %_1 to {}* 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         121: ; call sanitizer_kcfi_emit_type_metadata_trait_objects::foo3 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         122:  call void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo317hdb324cf16b626e72E({}* align 1 %_3.0, [3 x i64]* align 8 bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.2 to [3 x i64]*)) #2 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         123:  %b.0 = bitcast %Type3* %_1 to {}* 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         124:  %0 = load void ({}*, %Type3*)*, void ({}*, %Type3*)** getelementptr inbounds (void ({}*, %Type3*)*, void ({}*, %Type3*)** bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.2 to void ({}*, %Type3*)**), i64 3), align 8, !invariant.load !5, !nonnull !5 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         125:  call void %0({}* align 1 %b.0, %Type3* align 1 %_1) #2 [ "kcfi"(i32 -1903783964) ] 
         126:  ret void 
check:108     ~~~~~~~~~~
         127: } 
check:108     ~~
check:108     ~~
         128:  
check:108     ~
         129: attributes #0 = { inlinehint noredzone nounwind "target-cpu"="generic" "target-features"="+v8a,+strict-align,+neon,+fp-armv8" } 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         130: attributes #1 = { noredzone nounwind "target-cpu"="generic" "target-features"="+v8a,+strict-align,+neon,+fp-armv8" } 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         131: attributes #2 = { nounwind } 
         132:  
check:108     ~
check:108     ~
         133: !llvm.module.flags = !{!0} 
         134:  
check:108     ~
check:108     ~
         135: !0 = !{i32 4, !"kcfi", i32 1} 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         136: !1 = !{i32 -1522505972} 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~
         137: !2 = !{i32 -56908610} 
check:108     ~~~~~~~~~~~~~~~~~~~~~~
         138: !3 = !{i32 102973893} 
check:108     ~~~~~~~~~~~~~~~~~~~~~~
         139: !4 = !{i32 -1903783964} 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~
         140: !5 = !{} 
check:108     ~~~~~~~~~
         141: !6 = !{i32 -1134101214} 
>>>>>>
------------------------------------------



---- [codegen] tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs#x86_64 stdout ----

error in revision `x86_64`: verification with 'FileCheck' failed
status: exit status: 1
command: "/usr/lib/llvm-14/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.x86_64/sanitizer-kcfi-emit-type-metadata-trait-objects.ll" "/checkout/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs" "--allow-unused-prefixes" "--check-prefixes" "CHECK,NONMSVC,x86_64" "--dump-input-context" "100"
stdout: none
/checkout/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs:99:12: error: CHECK: expected string not found in input
/checkout/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs:99:12: error: CHECK: expected string not found in input
 // CHECK: call void %{{[0-9]}}({{\{\}\*|ptr}} align 1 {{%[a-z]\.0|%_[0-9]}}, {{\{\}\*|ptr}} align 1 {{%[a-z]\.0|%_[0-9]}}){{.*}}[ "kcfi"(i32 [[TYPE3:[[:print:]]+]]) ]
           ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.x86_64/sanitizer-kcfi-emit-type-metadata-trait-objects.ll:105:173: note: scanning from here
define void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo317hdb324cf16b626e72E({}* align 1 %a.0, [3 x i64]* align 8 %a.1) unnamed_addr #1 !<unknown kind #36> !9 {
/checkout/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs:108:12: error: CHECK: expected string not found in input
/checkout/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs:108:12: error: CHECK: expected string not found in input
 // CHECK: call void %{{[0-9]}}({{\{\}\*|ptr}} align 1 {{%[a-z]\.0|%_[0-9]}}, {{\{\}\*|ptr}} align 1 {{%[a-z]\.0|%_[0-9]}}){{.*}}[ "kcfi"(i32 [[TYPE3:[[:print:]]+]]) ]
           ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.x86_64/sanitizer-kcfi-emit-type-metadata-trait-objects.ll:117:132: note: scanning from here
define void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4bar317hd298b3b9b9d428c0E() unnamed_addr #1 !<unknown kind #36> !4 {


Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.x86_64/sanitizer-kcfi-emit-type-metadata-trait-objects.ll


-dump-input=help explains the following input dump.
Input was:
<<<<<<
           .
           .
           .
           .
           5:  
           6: %Type1 = type {} 
           7: %Type2 = type {} 
           8: %Type3 = type {} 
           9:  
          10: @vtable.0 = private unnamed_addr constant <{ i8*, [16 x i8], i8* }> <{ i8* bitcast (void ()* @"_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects78drop_in_place_fn$LT$sanitizer_kcfi_emit_type_metadata_trait_objects..Type1$GT$17hd544d746556e563bE" to i8*), [16 x i8] c"\00\00\00\00\00\00\00\00\01\00\00\00\00\00\00\00", i8* bitcast (void (%Type1*)* @"_ZN130_$LT$sanitizer_kcfi_emit_type_metadata_trait_objects..Type1$u20$as$u20$sanitizer_kcfi_emit_type_metadata_trait_objects..Trait1$GT$3foo17h3ed069aedb940e54E" to i8*) }>, align 8 
          11: @vtable.1 = private unnamed_addr constant <{ i8*, [16 x i8], i8* }> <{ i8* bitcast (void ()* @"_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects78drop_in_place_fn$LT$sanitizer_kcfi_emit_type_metadata_trait_objects..Type2$GT$17h8aa09b16cfc8ee65E" to i8*), [16 x i8] c"\00\00\00\00\00\00\00\00\01\00\00\00\00\00\00\00", i8* bitcast (void (%Type2*)* @"_ZN141_$LT$sanitizer_kcfi_emit_type_metadata_trait_objects..Type2$u20$as$u20$sanitizer_kcfi_emit_type_metadata_trait_objects..Trait2$LT$i32$GT$$GT$3bar17hc43b8b747eac5e80E" to i8*) }>, align 8 
          12: @vtable.2 = private unnamed_addr constant <{ i8*, [16 x i8], i8* }> <{ i8* bitcast (void ()* @"_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects78drop_in_place_fn$LT$sanitizer_kcfi_emit_type_metadata_trait_objects..Type3$GT$17h3b9500d3f54315f5E" to i8*), [16 x i8] c"\00\00\00\00\00\00\00\00\01\00\00\00\00\00\00\00", i8* bitcast (void (%Type3*, %Type3*)* @"_ZN86_$LT$T$u20$as$u20$sanitizer_kcfi_emit_type_metadata_trait_objects..Trait3$LT$U$GT$$GT$3baz17h3db20bfe4fcf6962E" to i8*) }>, align 8 
          13:  
          14: ; sanitizer_kcfi_emit_type_metadata_trait_objects::drop_in_place_fn<sanitizer_kcfi_emit_type_metadata_trait_objects::Type1> 
          15: ; Function Attrs: inlinehint noredzone nounwind nonlazybind 
          16: define internal void @"_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects78drop_in_place_fn$LT$sanitizer_kcfi_emit_type_metadata_trait_objects..Type1$GT$17hd544d746556e563bE"() unnamed_addr #0 !<unknown kind #36> !4 { 
          18:  ret void 
          19: } 
          20:  
          20:  
          21: ; sanitizer_kcfi_emit_type_metadata_trait_objects::drop_in_place_fn<sanitizer_kcfi_emit_type_metadata_trait_objects::Type2> 
          22: ; Function Attrs: inlinehint noredzone nounwind nonlazybind 
          23: define internal void @"_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects78drop_in_place_fn$LT$sanitizer_kcfi_emit_type_metadata_trait_objects..Type2$GT$17h8aa09b16cfc8ee65E"() unnamed_addr #0 !<unknown kind #36> !4 { 
          25:  ret void 
          26: } 
          27:  
          27:  
          28: ; sanitizer_kcfi_emit_type_metadata_trait_objects::drop_in_place_fn<sanitizer_kcfi_emit_type_metadata_trait_objects::Type3> 
          29: ; Function Attrs: inlinehint noredzone nounwind nonlazybind 
          30: define internal void @"_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects78drop_in_place_fn$LT$sanitizer_kcfi_emit_type_metadata_trait_objects..Type3$GT$17h3b9500d3f54315f5E"() unnamed_addr #0 !<unknown kind #36> !4 { 
          32:  ret void 
          33: } 
          34:  
          34:  
          35: ; <sanitizer_kcfi_emit_type_metadata_trait_objects::Type1 as sanitizer_kcfi_emit_type_metadata_trait_objects::Trait1>::foo 
          36: ; Function Attrs: noredzone nounwind nonlazybind 
          37: define void @"_ZN130_$LT$sanitizer_kcfi_emit_type_metadata_trait_objects..Type1$u20$as$u20$sanitizer_kcfi_emit_type_metadata_trait_objects..Trait1$GT$3foo17h3ed069aedb940e54E"(%Type1* align 1 %self) unnamed_addr #1 !<unknown kind #36> !5 { 
          39:  ret void 
          40: } 
          41:  
          41:  
          42: ; <sanitizer_kcfi_emit_type_metadata_trait_objects::Type2 as sanitizer_kcfi_emit_type_metadata_trait_objects::Trait2<i32>>::bar 
          43: ; Function Attrs: noredzone nounwind nonlazybind 
          44: define void @"_ZN141_$LT$sanitizer_kcfi_emit_type_metadata_trait_objects..Type2$u20$as$u20$sanitizer_kcfi_emit_type_metadata_trait_objects..Trait2$LT$i32$GT$$GT$3bar17hc43b8b747eac5e80E"(%Type2* align 1 %self) unnamed_addr #1 !<unknown kind #36> !6 { 
          46:  ret void 
          47: } 
          48:  
          48:  
          49: ; <T as sanitizer_kcfi_emit_type_metadata_trait_objects::Trait3<U>>::baz 
          50: ; Function Attrs: noredzone nounwind nonlazybind 
          51: define void @"_ZN86_$LT$T$u20$as$u20$sanitizer_kcfi_emit_type_metadata_trait_objects..Trait3$LT$U$GT$$GT$3baz17h3db20bfe4fcf6962E"(%Type3* align 1 %self, %Type3* align 1 %_2) unnamed_addr #1 !<unknown kind #36> !7 { 
          53:  ret void 
          54: } 
          55:  
          55:  
          56: ; sanitizer_kcfi_emit_type_metadata_trait_objects::foo1 
          57: ; Function Attrs: noredzone nounwind nonlazybind 
          58: define void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo117h93502349188b3e24E({}* align 1 %a.0, [3 x i64]* align 8 %a.1) unnamed_addr #1 !<unknown kind #36> !5 { 
          59: start: 
          60:  %0 = bitcast [3 x i64]* %a.1 to void ({}*)** 
          61:  %1 = getelementptr inbounds void ({}*)*, void ({}*)** %0, i64 3 
          62:  %2 = load void ({}*)*, void ({}*)** %1, align 8, !invariant.load !8, !nonnull !8 
          63:  call void %2({}* align 1 %a.0) #2 [ "kcfi"(i32 -56908610) ] 
          64:  ret void 
          65: } 
          66:  
          67: ; sanitizer_kcfi_emit_type_metadata_trait_objects::bar1 
          68: ; Function Attrs: noredzone nounwind nonlazybind 
          69: define void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4bar117had3ab10c0b90936dE() unnamed_addr #1 !<unknown kind #36> !4 { 
          70: start: 
          71:  %_1 = alloca %Type1, align 1 
          72:  %b.0 = bitcast %Type1* %_1 to {}* 
          73:  %0 = load void ({}*)*, void ({}*)** getelementptr inbounds (void ({}*)*, void ({}*)** bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.0 to void ({}*)**), i64 3), align 8, !invariant.load !8, !nonnull !8 
          74:  call void %0({}* align 1 %b.0) #2 [ "kcfi"(i32 -56908610) ] 
          75:  ret void 
          76: } 
          77:  
          78: ; sanitizer_kcfi_emit_type_metadata_trait_objects::foo2 
          79: ; Function Attrs: noredzone nounwind nonlazybind 
          80: define void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo217h8161e42dd83c0851E({}* align 1 %a.0, [3 x i64]* align 8 %a.1) unnamed_addr #1 !<unknown kind #36> !6 { 
          81: start: 
          82:  %0 = bitcast [3 x i64]* %a.1 to void ({}*)** 
          83:  %1 = getelementptr inbounds void ({}*)*, void ({}*)** %0, i64 3 
          84:  %2 = load void ({}*)*, void ({}*)** %1, align 8, !invariant.load !8, !nonnull !8 
          85:  call void %2({}* align 1 %a.0) #2 [ "kcfi"(i32 102973893) ] 
          86:  ret void 
          87: } 
          88:  
          89: ; sanitizer_kcfi_emit_type_metadata_trait_objects::bar2 
          90: ; Function Attrs: noredzone nounwind nonlazybind 
          91: define void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4bar217h18ebf9f7b2e09ed4E() unnamed_addr #1 !<unknown kind #36> !4 { 
          92: start: 
          93:  %_1 = alloca %Type2, align 1 
          94:  %_3.0 = bitcast %Type2* %_1 to {}* 
          95: ; call sanitizer_kcfi_emit_type_metadata_trait_objects::foo2 
          96:  call void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo217h8161e42dd83c0851E({}* align 1 %_3.0, [3 x i64]* align 8 bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.1 to [3 x i64]*)) #2 
          97:  %b.0 = bitcast %Type2* %_1 to {}* 
          98:  %0 = load void ({}*)*, void ({}*)** getelementptr inbounds (void ({}*)*, void ({}*)** bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.1 to void ({}*)**), i64 3), align 8, !invariant.load !8, !nonnull !8 
          99:  call void %0({}* align 1 %b.0) #2 [ "kcfi"(i32 102973893) ] 
         100:  ret void 
         101: } 
         102:  
         103: ; sanitizer_kcfi_emit_type_metadata_trait_objects::foo3 
         104: ; Function Attrs: noredzone nounwind nonlazybind 
         105: define void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo317hdb324cf16b626e72E({}* align 1 %a.0, [3 x i64]* align 8 %a.1) unnamed_addr #1 !<unknown kind #36> !9 { 
check:99                                                                                                                                                                                  X~~ error: no match found
check:99      ~~~~~~~
check:99      ~~~~~~~
         107:  %_2 = alloca %Type3, align 1 
check:99      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         108:  %0 = bitcast [3 x i64]* %a.1 to void ({}*, %Type3*)** 
check:99      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         109:  %1 = getelementptr inbounds void ({}*, %Type3*)*, void ({}*, %Type3*)** %0, i64 3 
check:99      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         110:  %2 = load void ({}*, %Type3*)*, void ({}*, %Type3*)** %1, align 8, !invariant.load !8, !nonnull !8 
check:99      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         111:  call void %2({}* align 1 %a.0, %Type3* align 1 %_2) #2 [ "kcfi"(i32 -1903783964) ] 
         112:  ret void 
check:99      ~~~~~~~~~~
         113: } 
check:99      ~~
check:99      ~~
         114:  
check:99      ~
         115: ; sanitizer_kcfi_emit_type_metadata_trait_objects::bar3 
check:99      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         116: ; Function Attrs: noredzone nounwind nonlazybind 
check:99      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         117: define void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4bar317hd298b3b9b9d428c0E() unnamed_addr #1 !<unknown kind #36> !4 { 
check:99      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:108                                                                                                                                        X~~ error: no match found
check:108     ~~~~~~~
check:108     ~~~~~~~
         119:  %_1 = alloca %Type3, align 1 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         120:  %_3.0 = bitcast %Type3* %_1 to {}* 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         121: ; call sanitizer_kcfi_emit_type_metadata_trait_objects::foo3 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         122:  call void @_ZN47sanitizer_kcfi_emit_type_metadata_trait_objects4foo317hdb324cf16b626e72E({}* align 1 %_3.0, [3 x i64]* align 8 bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.2 to [3 x i64]*)) #2 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         123:  %b.0 = bitcast %Type3* %_1 to {}* 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         124:  %0 = load void ({}*, %Type3*)*, void ({}*, %Type3*)** getelementptr inbounds (void ({}*, %Type3*)*, void ({}*, %Type3*)** bitcast (<{ i8*, [16 x i8], i8* }>* @vtable.2 to void ({}*, %Type3*)**), i64 3), align 8, !invariant.load !8, !nonnull !8 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         125:  call void %0({}* align 1 %b.0, %Type3* align 1 %_1) #2 [ "kcfi"(i32 -1903783964) ] 
         126:  ret void 
check:108     ~~~~~~~~~~
         127: } 
check:108     ~~
check:108     ~~
         128:  
check:108     ~
         129: attributes #0 = { inlinehint noredzone nounwind nonlazybind "probe-stack"="__rust_probestack" "target-cpu"="x86-64" "target-features"="-mmx,-sse,-sse2,-sse3,-ssse3,-sse4.1,-sse4.2,-3dnow,-3dnowa,-avx,-avx2,+soft-float" } 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         130: attributes #1 = { noredzone nounwind nonlazybind "probe-stack"="__rust_probestack" "target-cpu"="x86-64" "target-features"="-mmx,-sse,-sse2,-sse3,-ssse3,-sse4.1,-sse4.2,-3dnow,-3dnowa,-avx,-avx2,+soft-float" } 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         131: attributes #2 = { nounwind } 
         132:  
check:108     ~
check:108     ~
         133: !llvm.module.flags = !{!0, !1, !2, !3} 
         134:  
check:108     ~
check:108     ~
         135: !0 = !{i32 7, !"PIC Level", i32 2} 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         136: !1 = !{i32 1, !"Code Model", i32 2} 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         137: !2 = !{i32 2, !"RtLibUseGOT", i32 1} 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         138: !3 = !{i32 4, !"kcfi", i32 1} 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         139: !4 = !{i32 -1522505972} 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~
         140: !5 = !{i32 -56908610} 
check:108     ~~~~~~~~~~~~~~~~~~~~~~
         141: !6 = !{i32 102973893} 
check:108     ~~~~~~~~~~~~~~~~~~~~~~
         142: !7 = !{i32 -1903783964} 
check:108     ~~~~~~~~~~~~~~~~~~~~~~~~
         143: !8 = !{} 
check:108     ~~~~~~~~~
         144: !9 = !{i32 -1134101214} 
>>>>>>
------------------------------------------


```

</details>

--- a/compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs
+++ b/compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs
@@ -673,6 +673,14 @@ fn encode_ty<'tcx>(
             typeid.push_str(&s);
         }
 
+        // Type parameters
+        ty::Param(..) => {
+            // u5param as vendor extended type
+            let mut s = String::from("u5param");
+            compress(dict, DictKey::Ty(ty, TyQ::None), &mut s);
+            typeid.push_str(&s);
+        }
+
         // Unexpected types
         ty::Bound(..)
         | ty::Error(..)
@@ -680,13 +688,47 @@ fn encode_ty<'tcx>(
         | ty::GeneratorWitnessMIR(..)
         | ty::Infer(..)
         | ty::Alias(..)
-        | ty::Param(..)
         | ty::Placeholder(..) => {
             bug!("encode_ty: unexpected `{:?}`", ty.kind());
         }
     };
 
     typeid
+}
+
+/// Transforms predicates for being encoded and used in the substitution dictionary.
+fn transform_predicates<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    predicates: &List<ty::PolyExistentialPredicate<'tcx>>,
+    _options: EncodeTyOptions,
+) -> &'tcx List<ty::PolyExistentialPredicate<'tcx>> {
+    let predicates: Vec<ty::PolyExistentialPredicate<'tcx>> = predicates
+        .iter()
+        .map(|predicate| match predicate.skip_binder() {
+            ty::ExistentialPredicate::Trait(trait_ref) => {
+                let trait_ref = ty::TraitRef::identity(tcx, trait_ref.def_id);
+                ty::Binder::dummy(ty::ExistentialPredicate::Trait(
+                    ty::ExistentialTraitRef::erase_self_ty(tcx, trait_ref),
+                ))
+            }
+            _ => predicate,
+        })
+        .collect();
+    tcx.mk_poly_existential_predicates(&predicates)
+}
+
+/// Transforms substs for being encoded and used in the substitution dictionary.
+fn transform_substs<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    substs: SubstsRef<'tcx>,
+    options: TransformTyOptions,
+) -> SubstsRef<'tcx> {
+    let substs = substs.iter().map(|subst| match subst.unpack() {
+        GenericArgKind::Type(ty) if ty.is_c_void(tcx) => tcx.mk_unit().into(),
+        GenericArgKind::Type(ty) => transform_ty(tcx, ty, options).into(),
+        _ => subst,
+    });
+    tcx.mk_substs_from_iter(substs)
 }
 
 // Transforms a ty:Ty for being encoded and used in the substitution dictionary. It transforms all
@@ -697,7 +739,7 @@ fn transform_ty<'tcx>(tcx: TyCtxt<'tcx>, ty: Ty<'tcx>, options: TransformTyOptio
     let mut ty = ty;
 
     match ty.kind() {
-        ty::Float(..) | ty::Char | ty::Str | ty::Never | ty::Foreign(..) | ty::Dynamic(..) => {}
+        ty::Float(..) | ty::Char | ty::Str | ty::Never | ty::Foreign(..) => {}
 
         ty::Bool => {
             if options.contains(EncodeTyOptions::NORMALIZE_INTEGERS) {
@@ -870,6 +912,14 @@ fn transform_ty<'tcx>(tcx: TyCtxt<'tcx>, ty: Ty<'tcx>, options: TransformTyOptio
             }
         }
 
+        ty::Dynamic(predicates, _region, kind) => {
+            ty = tcx.mk_dynamic(
+                transform_predicates(tcx, predicates, options),
+                tcx.lifetimes.re_erased,
+                *kind,
+            );
+        }
+
         ty::Bound(..)
         | ty::Error(..)
         | ty::GeneratorWitness(..)
@@ -883,20 +933,6 @@ fn transform_ty<'tcx>(tcx: TyCtxt<'tcx>, ty: Ty<'tcx>, options: TransformTyOptio
     }
 
     ty
-}
-
-/// Transforms substs for being encoded and used in the substitution dictionary.
-fn transform_substs<'tcx>(
-    tcx: TyCtxt<'tcx>,
-    substs: SubstsRef<'tcx>,
-    options: TransformTyOptions,
-) -> SubstsRef<'tcx> {
-    let substs = substs.iter().map(|subst| match subst.unpack() {
-        GenericArgKind::Type(ty) if ty.is_c_void(tcx) => tcx.mk_unit().into(),
-        GenericArgKind::Type(ty) => transform_ty(tcx, ty, options).into(),
-        _ => subst,
-    });
-    tcx.mk_substs_from_iter(substs)
 }
 
 /// Returns a type metadata identifier for the specified FnAbi using the Itanium C++ ABI with vendor

--- a/tests/codegen/sanitizer-cfi-emit-type-metadata-id-itanium-cxx-abi.rs
+++ b/tests/codegen/sanitizer-cfi-emit-type-metadata-id-itanium-cxx-abi.rs
@@ -44,7 +44,7 @@ impl<T> Trait1<T> for i32 {
 }
 
 // Trait implementation
-impl<T> Trait1<T> for Struct1<T> {
+impl<T, U> Trait1<T> for Struct1<U> {
     fn foo(&self) { }
 }
 
@@ -536,15 +536,15 @@ pub fn foo149(_: Type14<Bar>, _: Type14<Bar>, _: Type14<Bar>) { }
 // CHECK: ![[TYPE93]] = !{i64 0, !"_ZTSFvPFu3i32S_EE"}
 // CHECK: ![[TYPE94]] = !{i64 0, !"_ZTSFvPFu3i32S_ES0_E"}
 // CHECK: ![[TYPE95]] = !{i64 0, !"_ZTSFvPFu3i32S_ES0_S0_E"}
-// CHECK: ![[TYPE96]] = !{i64 0, !"_ZTSFvu3refIu3dynIu{{[0-9]+}}NtNtNtC{{[[:print:]]+}}_4core3ops8function2FnIu5tupleIu3i32EEu{{[0-9]+}}NtNtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnce6OutputIS0_ES_u6regionEEE"}
-// CHECK: ![[TYPE97]] = !{i64 0, !"_ZTSFvu3refIu3dynIu{{[0-9]+}}NtNtNtC{{[[:print:]]+}}_4core3ops8function2FnIu5tupleIu3i32EEu{{[0-9]+}}NtNtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnce6OutputIS0_ES_u6regionEES5_E"}
-// CHECK: ![[TYPE98]] = !{i64 0, !"_ZTSFvu3refIu3dynIu{{[0-9]+}}NtNtNtC{{[[:print:]]+}}_4core3ops8function2FnIu5tupleIu3i32EEu{{[0-9]+}}NtNtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnce6OutputIS0_ES_u6regionEES5_S5_E"}
-// CHECK: ![[TYPE99]] = !{i64 0, !"_ZTSFvu3refIu3dynIu{{[0-9]+}}NtNtNtC{{[[:print:]]+}}_4core3ops8function5FnMutIu5tupleIu3i32EEu{{[0-9]+}}NtNtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnce6OutputIS0_ES_u6regionEEE"}
-// CHECK: ![[TYPE100]] = !{i64 0, !"_ZTSFvu3refIu3dynIu{{[0-9]+}}NtNtNtC{{[[:print:]]+}}_4core3ops8function5FnMutIu5tupleIu3i32EEu{{[0-9]+}}NtNtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnce6OutputIS0_ES_u6regionEES5_E"}
-// CHECK: ![[TYPE101]] = !{i64 0, !"_ZTSFvu3refIu3dynIu{{[0-9]+}}NtNtNtC{{[[:print:]]+}}_4core3ops8function5FnMutIu5tupleIu3i32EEu{{[0-9]+}}NtNtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnce6OutputIS0_ES_u6regionEES5_S5_E"}
-// CHECK: ![[TYPE102]] = !{i64 0, !"_ZTSFvu3refIu3dynIu{{[0-9]+}}NtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnceIu5tupleIu3i32EEu{{[0-9]+}}NtNtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnce6OutputIS0_ES_u6regionEEE"}
-// CHECK: ![[TYPE103]] = !{i64 0, !"_ZTSFvu3refIu3dynIu{{[0-9]+}}NtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnceIu5tupleIu3i32EEu{{[0-9]+}}NtNtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnce6OutputIS0_ES_u6regionEES5_E"}
-// CHECK: ![[TYPE104]] = !{i64 0, !"_ZTSFvu3refIu3dynIu{{[0-9]+}}NtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnceIu5tupleIu3i32EEu{{[0-9]+}}NtNtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnce6OutputIS0_ES_u6regionEES5_S5_E"}
+// CHECK: ![[TYPE96]] = !{i64 0, !"_ZTSFvu3refIu3dynIu{{[0-9]+}}NtNtNtC{{[[:print:]]+}}_4core3ops8function2FnIu5paramEu{{[0-9]+}}NtNtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnce6OutputIu5tupleIu3i32EES1_u6regionEEE"}
+// CHECK: ![[TYPE97]] = !{i64 0, !"_ZTSFvu3refIu3dynIu{{[0-9]+}}NtNtNtC{{[[:print:]]+}}_4core3ops8function2FnIu5paramEu{{[0-9]+}}NtNtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnce6OutputIu5tupleIu3i32EES1_u6regionEES6_E"}
+// CHECK: ![[TYPE98]] = !{i64 0, !"_ZTSFvu3refIu3dynIu{{[0-9]+}}NtNtNtC{{[[:print:]]+}}_4core3ops8function2FnIu5paramEu{{[0-9]+}}NtNtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnce6OutputIu5tupleIu3i32EES1_u6regionEES6_S6_E"}
+// CHECK: ![[TYPE99]] = !{i64 0, !"_ZTSFvu3refIu3dynIu{{[0-9]+}}NtNtNtC{{[[:print:]]+}}_4core3ops8function5FnMutIu5paramEu{{[0-9]+}}NtNtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnce6OutputIu5tupleIu3i32EES1_u6regionEEE"}
+// CHECK: ![[TYPE100]] = !{i64 0, !"_ZTSFvu3refIu3dynIu{{[0-9]+}}NtNtNtC{{[[:print:]]+}}_4core3ops8function5FnMutIu5paramEu{{[0-9]+}}NtNtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnce6OutputIu5tupleIu3i32EES1_u6regionEES6_E"}
+// CHECK: ![[TYPE101]] = !{i64 0, !"_ZTSFvu3refIu3dynIu{{[0-9]+}}NtNtNtC{{[[:print:]]+}}_4core3ops8function5FnMutIu5paramEu{{[0-9]+}}NtNtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnce6OutputIu5tupleIu3i32EES1_u6regionEES6_S6_E"}
+// CHECK: ![[TYPE102]] = !{i64 0, !"_ZTSFvu3refIu3dynIu{{[0-9]+}}NtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnceIu5paramEu{{[0-9]+}}NtNtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnce6OutputIu5tupleIu3i32EES1_u6regionEEE"}
+// CHECK: ![[TYPE103]] = !{i64 0, !"_ZTSFvu3refIu3dynIu{{[0-9]+}}NtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnceIu5paramEu{{[0-9]+}}NtNtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnce6OutputIu5tupleIu3i32EES1_u6regionEES6_E"}
+// CHECK: ![[TYPE104]] = !{i64 0, !"_ZTSFvu3refIu3dynIu{{[0-9]+}}NtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnceIu5paramEu{{[0-9]+}}NtNtNtNtC{{[[:print:]]+}}_4core3ops8function6FnOnce6OutputIu5tupleIu3i32EES1_u6regionEES6_S6_E"}
 // CHECK: ![[TYPE105]] = !{i64 0, !"_ZTSFvu3refIu3dynIu{{[0-9]+}}NtNtC{{[[:print:]]+}}_4core6marker4Sendu6regionEEE"}
 // CHECK: ![[TYPE106]] = !{i64 0, !"_ZTSFvu3refIu3dynIu{{[0-9]+}}NtNtC{{[[:print:]]+}}_4core6marker4Sendu6regionEES2_E"}
 // CHECK: ![[TYPE107]] = !{i64 0, !"_ZTSFvu3refIu3dynIu{{[0-9]+}}NtNtC{{[[:print:]]+}}_4core6marker4Sendu6regionEES2_S2_E"}
@@ -566,9 +566,9 @@ pub fn foo149(_: Type14<Bar>, _: Type14<Bar>, _: Type14<Bar>) { }
 // CHECK: ![[TYPE123]] = !{i64 0, !"_ZTSFvu{{[0-9]+}}NvNINvC{{[[:print:]]+}}_51sanitizer_cfi_emit_type_metadata_id_itanium_cxx_abi3fn18{{[{}][{}]}}impl{{[}][}]}}3fooIu3i32EE"}
 // CHECK: ![[TYPE124]] = !{i64 0, !"_ZTSFvu{{[0-9]+}}NvNINvC{{[[:print:]]+}}_51sanitizer_cfi_emit_type_metadata_id_itanium_cxx_abi3fn18{{[{}][{}]}}impl{{[}][}]}}3fooIu3i32ES0_E"}
 // CHECK: ![[TYPE125]] = !{i64 0, !"_ZTSFvu{{[0-9]+}}NvNINvC{{[[:print:]]+}}_51sanitizer_cfi_emit_type_metadata_id_itanium_cxx_abi3fn18{{[{}][{}]}}impl{{[}][}]}}3fooIu3i32ES0_S0_E"}
-// CHECK: ![[TYPE126]] = !{i64 0, !"_ZTSFvu{{[0-9]+}}NvNtC{{[[:print:]]+}}_51sanitizer_cfi_emit_type_metadata_id_itanium_cxx_abi6Trait13fooIu3dynIu{{[0-9]+}}NtC{{[[:print:]]+}}_51sanitizer_cfi_emit_type_metadata_id_itanium_cxx_abi6Trait1Iu3i32Eu6regionES_EE"}
-// CHECK: ![[TYPE127]] = !{i64 0, !"_ZTSFvu{{[0-9]+}}NvNtC{{[[:print:]]+}}_51sanitizer_cfi_emit_type_metadata_id_itanium_cxx_abi6Trait13fooIu3dynIu{{[0-9]+}}NtC{{[[:print:]]+}}_51sanitizer_cfi_emit_type_metadata_id_itanium_cxx_abi6Trait1Iu3i32Eu6regionES_ES3_E"}
-// CHECK: ![[TYPE128]] = !{i64 0, !"_ZTSFvu{{[0-9]+}}NvNtC{{[[:print:]]+}}_51sanitizer_cfi_emit_type_metadata_id_itanium_cxx_abi6Trait13fooIu3dynIu{{[0-9]+}}NtC{{[[:print:]]+}}_51sanitizer_cfi_emit_type_metadata_id_itanium_cxx_abi6Trait1Iu3i32Eu6regionES_ES3_S3_E"}
+// CHECK: ![[TYPE126]] = !{i64 0, !"_ZTSFvu{{[0-9]+}}NvNtC{{[[:print:]]+}}_51sanitizer_cfi_emit_type_metadata_id_itanium_cxx_abi6Trait13fooIu3dynIu{{[0-9]+}}NtC{{[[:print:]]+}}_51sanitizer_cfi_emit_type_metadata_id_itanium_cxx_abi6Trait1Iu5paramEu6regionEu3i32EE"}
+// CHECK: ![[TYPE127]] = !{i64 0, !"_ZTSFvu{{[0-9]+}}NvNtC{{[[:print:]]+}}_51sanitizer_cfi_emit_type_metadata_id_itanium_cxx_abi6Trait13fooIu3dynIu{{[0-9]+}}NtC{{[[:print:]]+}}_51sanitizer_cfi_emit_type_metadata_id_itanium_cxx_abi6Trait1Iu5paramEu6regionEu3i32ES4_E"}
+// CHECK: ![[TYPE128]] = !{i64 0, !"_ZTSFvu{{[0-9]+}}NvNtC{{[[:print:]]+}}_51sanitizer_cfi_emit_type_metadata_id_itanium_cxx_abi6Trait13fooIu3dynIu{{[0-9]+}}NtC{{[[:print:]]+}}_51sanitizer_cfi_emit_type_metadata_id_itanium_cxx_abi6Trait1Iu5paramEu6regionEu3i32ES4_S4_E"}
 // CHECK: ![[TYPE129]] = !{i64 0, !"_ZTSFvu{{[0-9]+}}NvNtC{{[[:print:]]+}}_51sanitizer_cfi_emit_type_metadata_id_itanium_cxx_abi6Trait13fooIu3i32S_EE"}
 // CHECK: ![[TYPE130]] = !{i64 0, !"_ZTSFvu{{[0-9]+}}NvNtC{{[[:print:]]+}}_51sanitizer_cfi_emit_type_metadata_id_itanium_cxx_abi6Trait13fooIu3i32S_ES0_E"}
 // CHECK: ![[TYPE131]] = !{i64 0, !"_ZTSFvu{{[0-9]+}}NvNtC{{[[:print:]]+}}_51sanitizer_cfi_emit_type_metadata_id_itanium_cxx_abi6Trait13fooIu3i32S_ES0_S0_E"}

--- a/tests/codegen/sanitizer-cfi-emit-type-metadata-trait-objects.rs
+++ b/tests/codegen/sanitizer-cfi-emit-type-metadata-trait-objects.rs
@@ -1,44 +1,89 @@
 // Verifies that type metadata identifiers for trait objects are emitted correctly.
 //
 // needs-sanitizer-cfi
-// compile-flags: -Clto -Cno-prepopulate-passes -Ctarget-feature=-crt-static -Zsanitizer=cfi
+// compile-flags: -Clto -Cno-prepopulate-passes -Copt-level=0 -Ctarget-feature=-crt-static -Zsanitizer=cfi
 
 #![crate_type="lib"]
 
-trait Trait1 {
+pub trait Trait1 {
     fn foo(&self);
 }
 
-struct Type1;
+#[derive(Clone, Copy)]
+pub struct Type1;
 
 impl Trait1 for Type1 {
     fn foo(&self) {
     }
 }
 
-pub fn foo() {
-    let a = Type1;
-    a.foo();
-    // CHECK-LABEL: define{{.*}}foo{{.*}}!type !{{[0-9]+}}
-    // CHECK:       call <sanitizer_cfi_emit_type_metadata_trait_objects::Type1 as sanitizer_cfi_emit_type_metadata_trait_objects::Trait1>::foo
+pub trait Trait2<T> {
+    fn bar(&self);
 }
 
-pub fn bar() {
+pub struct Type2;
+
+impl Trait2<i32> for Type2 {
+    fn bar(&self) {
+    }
+}
+
+pub trait Trait3<T> {
+    fn baz(&self, _: &T);
+}
+
+pub struct Type3;
+
+impl<T, U> Trait3<U> for T {
+    fn baz(&self, _: &U) {
+    }
+}
+
+pub fn foo1(a: &dyn Trait1) {
+    a.foo();
+    // CHECK-LABEL: define{{.*}}4foo1{{.*}}!type !{{[0-9]+}}
+    // CHECK:       call i1 @llvm.type.test({{i8\*|ptr}} {{%f|%[0-9]}}, metadata !"[[TYPE1:[[:print:]]+]]")
+}
+
+pub fn bar1() {
     let a = Type1;
     let b = &a as &dyn Trait1;
     b.foo();
-    // CHECK-LABEL: define{{.*}}bar{{.*}}!type !{{[0-9]+}}
-    // CHECK:       call i1 @llvm.type.test({{i8\*|ptr}} {{%f|%0|%1}}, metadata !"[[TYPE1:[[:print:]]+]]")
+    // CHECK-LABEL: define{{.*}}4bar1{{.*}}!type !{{[0-9]+}}
+    // CHECK:       call i1 @llvm.type.test({{i8\*|ptr}} {{%f|%[0-9]}}, metadata !"[[TYPE2:[[:print:]]+]]")
 }
 
-pub fn baz() {
-    let a = Type1;
-    let b = &a as &dyn Trait1;
-    a.foo();
-    b.foo();
-    // CHECK-LABEL: define{{.*}}baz{{.*}}!type !{{[0-9]+}}
-    // CHECK:       call <sanitizer_cfi_emit_type_metadata_trait_objects::Type1 as sanitizer_cfi_emit_type_metadata_trait_objects::Trait1>::foo
-    // CHECK:       call i1 @llvm.type.test({{i8\*|ptr}} {{%f|%0|%1}}, metadata !"[[TYPE1:[[:print:]]+]]")
+pub fn foo2<T>(a: &dyn Trait2<T>) {
+    a.bar();
+    // CHECK-LABEL: define{{.*}}4foo2{{.*}}!type !{{[0-9]+}}
+    // CHECK:       call i1 @llvm.type.test({{i8\*|ptr}} {{%f|%[0-9]}}, metadata !"[[TYPE2:[[:print:]]+]]")
+}
+
+pub fn bar2() {
+    let a = Type2;
+    foo2(&a);
+    let b = &a as &dyn Trait2<i32>;
+    b.bar();
+    // CHECK-LABEL: define{{.*}}4bar2{{.*}}!type !{{[0-9]+}}
+    // CHECK:       call i1 @llvm.type.test({{i8\*|ptr}} {{%f|%[0-9]}}, metadata !"[[TYPE2:[[:print:]]+]]")
+}
+
+pub fn foo3(a: &dyn Trait3<Type3>) {
+    let b = Type3;
+    a.baz(&b);
+    // CHECK-LABEL: define{{.*}}4foo3{{.*}}!type !{{[0-9]+}}
+    // CHECK:       call i1 @llvm.type.test({{i8\*|ptr}} {{%f|%[0-9]}}, metadata !"[[TYPE3:[[:print:]]+]]")
+}
+
+pub fn bar3() {
+    let a = Type3;
+    foo3(&a);
+    let b = &a as &dyn Trait3<Type3>;
+    b.baz(&a);
+    // CHECK-LABEL: define{{.*}}4bar3{{.*}}!type !{{[0-9]+}}
+    // CHECK:       call i1 @llvm.type.test({{i8\*|ptr}} {{%f|%[0-9]}}, metadata !"[[TYPE3:[[:print:]]+]]")
 }
 
 // CHECK: !{{[0-9]+}} = !{i64 0, !"[[TYPE1]]"}
+// CHECK: !{{[0-9]+}} = !{i64 0, !"[[TYPE2]]"}
+// CHECK: !{{[0-9]+}} = !{i64 0, !"[[TYPE3]]"}

--- a/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs
+++ b/tests/codegen/sanitizer-kcfi-emit-type-metadata-trait-objects.rs
@@ -30,40 +30,84 @@ trait Freeze { }
 #[lang="drop_in_place"]
 fn drop_in_place_fn<T>() { }
 
-trait Trait1 {
+pub trait Trait1 {
     fn foo(&self);
 }
 
-struct Type1;
+pub struct Type1;
 
 impl Trait1 for Type1 {
     fn foo(&self) {
     }
 }
 
-pub fn foo() {
-    let a = Type1;
-    a.foo();
-    // CHECK-LABEL: define{{.*}}foo{{.*}}!{{<unknown kind #36>|kcfi_type}} !{{[0-9]+}}
-    // CHECK:       call <sanitizer_kcfi_emit_type_metadata_trait_objects::Type1 as sanitizer_kcfi_emit_type_metadata_trait_objects::Trait1>::foo
+pub trait Trait2<T> {
+    fn bar(&self);
 }
 
-pub fn bar() {
+pub struct Type2;
+
+impl Trait2<i32> for Type2 {
+    fn bar(&self) {
+    }
+}
+
+pub trait Trait3<T> {
+    fn baz(&self, _: &T);
+}
+
+pub struct Type3;
+
+impl<T, U> Trait3<U> for T {
+    fn baz(&self, _: &U) {
+    }
+}
+
+pub fn foo1(a: &dyn Trait1) {
+    a.foo();
+    // CHECK-LABEL: define{{.*}}4foo1{{.*}}!{{<unknown kind #36>|kcfi_type}} !{{[0-9]+}}
+    // CHECK:       call void %{{[0-9]}}({{\{\}\*|ptr}} align 1 {{%[a-z]\.0|%_[0-9]}}){{.*}}[ "kcfi"(i32 [[TYPE1:[[:print:]]+]]) ]
+}
+
+pub fn bar1() {
     let a = Type1;
     let b = &a as &dyn Trait1;
     b.foo();
-    // CHECK-LABEL: define{{.*}}bar{{.*}}!{{<unknown kind #36>|kcfi_type}} !{{[0-9]+}}
-    // CHECK:       call void %0({{\{\}\*|ptr}} align 1 {{%b\.0|%_1}}){{.*}}[ "kcfi"(i32 [[TYPE1:[[:print:]]+]]) ]
+    // CHECK-LABEL: define{{.*}}4bar1{{.*}}!{{<unknown kind #36>|kcfi_type}} !{{[0-9]+}}
+    // CHECK:       call void %{{[0-9]}}({{\{\}\*|ptr}} align 1 {{%[a-z]\.0|%_[0-9]}}){{.*}}[ "kcfi"(i32 [[TYPE1:[[:print:]]+]]) ]
 }
 
-pub fn baz() {
-    let a = Type1;
-    let b = &a as &dyn Trait1;
-    a.foo();
-    b.foo();
-    // CHECK-LABEL: define{{.*}}baz{{.*}}!{{<unknown kind #36>|kcfi_type}} !{{[0-9]+}}
-    // CHECK:       call <sanitizer_kcfi_emit_type_metadata_trait_objects::Type1 as sanitizer_kcfi_emit_type_metadata_trait_objects::Trait1>::foo
-    // CHECK:       call void %0({{\{\}\*|ptr}} align 1 {{%b\.0|%_1}}){{.*}}[ "kcfi"(i32 [[TYPE1:[[:print:]]+]]) ]
+pub fn foo2<T>(a: &dyn Trait2<T>) {
+    a.bar();
+    // CHECK-LABEL: define{{.*}}4foo2{{.*}}!{{<unknown kind #36>|kcfi_type}} !{{[0-9]+}}
+    // CHECK:       call void %{{[0-9]}}({{\{\}\*|ptr}} align 1 {{%[a-z]\.0|%_[0-9]}}){{.*}}[ "kcfi"(i32 [[TYPE2:[[:print:]]+]]) ]
+}
+
+pub fn bar2() {
+    let a = Type2;
+    foo2(&a);
+    let b = &a as &dyn Trait2<i32>;
+    b.bar();
+    // CHECK-LABEL: define{{.*}}4bar2{{.*}}!{{<unknown kind #36>|kcfi_type}} !{{[0-9]+}}
+    // CHECK:       call void %{{[0-9]}}({{\{\}\*|ptr}} align 1 {{%[a-z]\.0|%_[0-9]}}){{.*}}[ "kcfi"(i32 [[TYPE2:[[:print:]]+]]) ]
+}
+
+pub fn foo3(a: &dyn Trait3<Type3>) {
+    let b = Type3;
+    a.baz(&b);
+    // CHECK-LABEL: define{{.*}}4foo3{{.*}}!{{<unknown kind #36>|kcfi_type}} !{{[0-9]+}}
+    // CHECK:       call void %{{[0-9]}}({{\{\}\*|ptr}} align 1 {{%[a-z]\.0|%_[0-9]}}, {{\{\}\*|ptr|%Type3\*}} align 1 {{%[a-z]\.0|%_[0-9]}}){{.*}}[ "kcfi"(i32 [[TYPE3:[[:print:]]+]]) ]
+}
+
+pub fn bar3() {
+    let a = Type3;
+    foo3(&a);
+    let b = &a as &dyn Trait3<Type3>;
+    b.baz(&a);
+    // CHECK-LABEL: define{{.*}}4bar3{{.*}}!{{<unknown kind #36>|kcfi_type}} !{{[0-9]+}}
+    // CHECK:       call void %{{[0-9]}}({{\{\}\*|ptr}} align 1 {{%[a-z]\.0|%_[0-9]}}, {{\{\}\*|ptr|%Type3\*}} align 1 {{%[a-z]\.0|%_[0-9]}}){{.*}}[ "kcfi"(i32 [[TYPE3:[[:print:]]+]]) ]
 }
 
 // CHECK: !{{[0-9]+}} = !{i32 [[TYPE1]]}
+// CHECK: !{{[0-9]+}} = !{i32 [[TYPE2]]}
+// CHECK: !{{[0-9]+}} = !{i32 [[TYPE3]]}

